readiness-tool/popup.js

```diff
@@ -0,0 +1,246 @@
+/**
+ * @fileoverview Entry point for ART Chrome extension. 
+ * Scans the current page against the vendors in the app.json file and
+ * specifies them as either supported or not supported.
+ */
+self.popups = {};
+self.popups.isSupported = isSupported;
+self.popups.addToDict = addToDict;
+/** @const {!Element} */
+let supportedAds;
+/* @const {!Element} */
+let supportedAnalytics;
+/** @const {!Element} */
+let notSupportedAds;
+/** @const {!Element} */
+let notSupportedAnalytics;
+/* @const {string} */
+const loadingMessage = 'Loading...';
+/** @const {string} */
+const blankMessage = '';
+/** @const {string} */
+const appsRegexDelimeter = '\\;';
+
+/**
+ * Callback function that sends a response upon receiving message
+ * @param {!Object} request - Message Object
+ * @param {!Object} sender - Message sender defined here
+ */
+chrome.runtime.onMessage.addListener(function (request, sender) {
+  if (request.action == 'getSource') {
+    // htmlOfTab contains the page source as a string
+    htmlOfTab = request.source;
+    findDetectedApps(htmlOfTab);
+  }
+});
+window.onload = function onWindowLoad() {
+  // When page is loaded, display 'Loading...' so the user expects content
+  supportedAds = document.getElementById('ads-supported');
+  supportedAnalytics = document.getElementById('analytics-supported');
+  notSupportedAds = document.getElementById('ads-notSupported');
+  notSupportedAnalytics = document.getElementById('analytics-notSupported');
+  supportedAds.textContent = supportedAnalytics.textContent =
+    notSupportedAds.textContent = notSupportedAnalytics.textContent =
+    loadingMessage;
+  // Gets the DOM of the current web page and converts it to a string
+  chrome.tabs.executeScript(null, {
+    file: 'getPagesSource.js',
+  }, function () {});
+};
+/**
+ * Returns all the 3rd party applications found on the website
+ * @param {string} html - String containing all HTML on the page
+ * @return {Object} 
+ */
+function findDetectedApps(html) {
+  fetch('apps.json').then(function (response) {
+      response.json().then(function (data) {
+        let listAllApps = data.apps;
+        detectedApps = filteredApps(html, listAllApps);
+        showSupportedAppsInView(detectedApps, listAllApps);
+        return detectedApps;
+      });
+    }).then(data => console.log('data is', data))
+    .catch(error => console.log('error is', error))
```

Comment:
Instead of chaining a `Promise#catch`, you can use the second arg of `Promise#then`:

```js
fetch(...).then(data => {
  console.log(...);
}, error => {
  console.error(...);
});
```